### DISCUSSION TITLE
Android Error on startScan method.

### DISCUSSION THREAD

**Discussion by CntDoit** [capawesome-team/capacitor-mlkit#24] [Q&A] [answered]
Hello! Thank you for this plugin, we've been waiting for something like this! We need to scan multiple code128 barcodes without the camera view closing.

So I followed your setup guide. but im getting an error on Android:
`Uncaught (in promise): Error: Attempt to invoke virtual method 'androidx.camera.core.Preview$SurfaceProvider androidx.camera.view.PreviewView.getSurfaceProvider()' on a null object reference`

I'm using the startScan method example:
```ts
 const startScan = async () => {
   document.querySelector('body')?.classList.add('barcode-scanner-active');

   const listener = await BarcodeScanner.addListener('barcodeScanned', async 
      result => { console.log(result.barcode); });

   await BarcodeScanner.startScan();
};
``` 

I'm getting this error when using startScan method. If i use the scan method (which would be ideal because it has the UI already made) this does not happen. It scans and closes out the camera view.
```ts   
const scan = async () => {
  const { barcodes } = await BarcodeScanner.scan({
    formats: [BarcodeFormat.Code128]
  });
  return barcodes;
};
``` 

How can i use the scan method to not close and keep scanning barcodes? 

Thanks in advance for your help.

**Comment by robingenz** [comment, marked as answer]
I'm very glad you found the plugin. I am sure we will find a solution.

Does this error also occur with the [ML Kit Demo App](https://github.com/robingenz/capacitor-mlkit-plugin-demo)?

**Reply by CntDoit** [reply]
Oh nice, i didnt know there was a demo app, I didnt see a link for it in the barcode plugin read.me file. Anyways, I tested it out and I clicked on start scan but theres just a black screen, and no errors show in console. please see quick video: https://recordit.co/6uUC3OpIrs

The scan options does work just like on my app. Not sure if this matters but on im a pixel 4a Android 13.

**Reply by robingenz** [reply]
> Oh nice, i didnt know there was a demo app, I didnt see a link for it in the barcode plugin read.me file. 

Oh, my bad. I will add it. Did you request permissions?

**Edit**: The `scan()` method does not require permissions, but the `startScan()` method does.

**Reply by CntDoit** [reply]
Update: so the black screen is because i needed to click on the request permissions first. It's showing now and it is scanning the barcode but it closes, right after. I will see what the demo app has done differently then my app. Thank you for the quick response and your help! Have a great day!

**Reply by robingenz** [reply]
Great, let me know if you have any other questions.
